util_for_test.go

```diff
@@ -0,0 +1,329 @@
+package surrealdb
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	rawslog "log/slog"
+	"net"
+	"os/exec"
+	"reflect"
+	"strings"
+	"sync"
+	"testing"
+	"time"
+
+	"github.com/surrealdb/surrealdb.go/pkg/conn/gorilla"
+	"github.com/surrealdb/surrealdb.go/pkg/logger/slog"
+)
+
+var (
+	testHasSurrealCLI     bool
+	checkSurrealCLIOnce   sync.Once
+	DelayAfterServerStart time.Duration = 300 * time.Millisecond
+	DelayBeforeServerExit time.Duration = 300 * time.Millisecond
+)
+
+func Test_NewTestSurrealDB_Simple(t *testing.T) {
+	// Simple illustration of how NewTestSurrealDB can be used.
+
+	// Tests can run in parallel.
+	t.Parallel()
+
+	// NewTestSurrealDB returns endpoint (random open port used), DB instance
+	// (which is wrapped in DBForTest to provide extra methods), and a func to
+	// shutdown the SurrealDB server.
+	endpoint, db, close := NewTestSurrealDB(t)
+	defer close()
+	_ = endpoint // endpoint could be useful for some low level testing.
+
+	// DBForTest.Prepare is to run any SurrealQL for test prep. This would fail
+	// early if the provided string results in an error from SurrealDB server.
```

Comment:
> Is the purpose to validate each individual response? Could call the helper function MustQuery then. It would follow the same structure as Query but would error or panic on failed asserts?

I believe we really shouldn't have a `MustQuery` func. Unlike regexp, the query may or may not fail due to the SurrealDB implementation, and is dependant on the environment setup (such as duplicated `CREATE` calls). The point here was so that `Prepare` would fail early by `t.Fatal`, and is meant to be only used for test code. Because the file ends with `_test.go`, this will not be a part of the SDK for other packages to consume.